### PR TITLE
Update spec for tkg vsphere workload cluster corresponding to changes in API

### PR DESCRIPTION
**Pull request by Vasundharashukla**
Signed-off-by: Vasundhara Shukla <vasundharas@vmware.com>

1. **What this PR does / why we need it**:
Add new spec attributes for TKG vSphere workload cluster.

2. **Additional information**
Added changes to acceptance test, example, resource templates and documentation.

### CODE REVIEW

File: internal/resources/cluster/common/advanced_configs.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+)
+
+const (
+	advancedConfigurationKey      = "key"
+	advancedConfigurationValueKey = "value"
+)
+
+var AdvancedConfigs = &schema.Schema{
+	Type:        schema.TypeList,
+	Description: "Advanced configuration for TKGm cluster",
```
Comment (shreyassreenivas) [inline internal/resources/cluster/common/advanced_configs.go:21]: advance configuration of what ?

**Review by shreyassreenivas — APPROVED** [review]

**Comment by shreyassreenivas** [comment]
Acceptance is pending right ?

File: internal/resources/cluster/common/advanced_configs.go (hunk 1 above)
Comment (Vasundharashukla) [reply internal/resources/cluster/common/advanced_configs.go:21]: There is a new spec in API for TKGm workload clusters called Advanced Config that has key and value in it. The Description is the same taken from API. It is extra configuration for the workload clusters.

Comment (shreyassreenivas) [reply internal/resources/cluster/common/advanced_configs.go:21]: if possible can you add more information on what is the advance configuration used for  ?

Comment (Vasundharashukla) [reply internal/resources/cluster/common/advanced_configs.go:21]: Not got any extra information in the API changes, even checked the PR, let me reach out to the concerned engineer who has done this addition.

File: docs/data-sources/cluster.md [hunk 2]
```diff
@@ -634,3 +644,16 @@ Optional:
 - `cpu` (String) Number of CPUs per node
 - `disk_size` (String) Root disk size in gigabytes for the VM
 - `memory` (String) Memory associated with the node in megabytes
+
+
+
+
+
+
```
Comment (shobha2626) [inline docs/data-sources/cluster.md:652]: nit: so many newlines?

**Review by shobha2626 — CHANGES_REQUESTED** [review]

**Review by ishangupta-ds — APPROVED** [review]
LGTM.

nit:
- Please add any required references to TMC documentation for this advanced config.
- Please remove the line breaks which are not internationally added.
- Please update the unit tests and acceptance tests.

**Comment by Vasundharashukla** [comment]
Waiting to get any examples that can be added to the documentation to make it more clear on the functionality of advanced configs. The description in the schema is sufficient and the same will be used to describe tf schema. 
cc: @shreyassreenivas

**Review by ishangupta-ds — APPROVED** [review]

**Review by shreyassreenivas — APPROVED** [review]

File: docs/data-sources/cluster.md (hunk 2 above)
Comment (shobha2626) [reply docs/data-sources/cluster.md:652]: @Vasundharashukla ^^

Comment (shobha2626) [reply docs/data-sources/cluster.md:652]: Generated as part of doc generation

**Pull request merged**